src/components/scenes/FioAddressRegisterScene.js

```diff
@@ -0,0 +1,200 @@
+// @flow
+
+import { type EdgeCurrencyWallet } from 'edge-core-js'
+import React, { Component } from 'react'
+import { Image, View } from 'react-native'
+
+import fioAddressDetailsIcon from '../../assets/images/details_fioAddress.png'
+import * as Constants from '../../constants/indexConstants'
+import { FIO_DOMAIN_DEFAULT, FIO_WALLET_TYPE } from '../../constants/WalletAndCurrencyConstants'
+import s from '../../locales/strings.js'
+import { Button } from '../../modules/UI/components/ControlPanel/Component/Button/Button.ui'
+import T from '../../modules/UI/components/FormattedText/index'
+import Gradient from '../../modules/UI/components/Gradient/Gradient.ui'
+import { Icon } from '../../modules/UI/components/Icon/Icon.ui'
+import SafeAreaView from '../../modules/UI/components/SafeAreaView/index'
+import { MaterialInputOnWhite } from '../../styles/components/FormFieldStyles.js'
+import styles from '../../styles/scenes/FioAddressRegisterStyle'
+import THEME from '../../theme/variables/airbitz'
+import type { CheckConnectivityProps, GoToSceneProps } from '../../types/types'
+import { validateFioAddress } from '../../util/utils'
+import { FormField } from '../common/FormField.js'
+
+export type State = {
+  fioAddress: string,
+  isValid: boolean,
+  touched: boolean,
+  isAvailable: boolean | null,
+  replaceRegex: RegExp
+}
+
+export type StateProps = {
+  fioWallets: EdgeCurrencyWallet[]
+}
+
+export type DispatchProps = {
+  changeFioAddressName(fioAddressName: string): any,
+  createCurrencyWallet: (walletName: string, walletType: string) => any
+}
+
+type Props = StateProps & DispatchProps & GoToSceneProps & CheckConnectivityProps
+
+export class FioAddressRegisterScene extends Component<Props, State> {
+  clearButtonMode = 'while-editing'
+  returnKeyType = 'next'
+  inputMaxLength = 15
+
+  state = {
+    fioAddress: '',
+    isValid: false,
+    touched: false,
+    isAvailable: false,
+    replaceRegex: new RegExp(`${FIO_DOMAIN_DEFAULT}`)
+  }
+
+  wallet: EdgeCurrencyWallet | null
+  bcValidationTimer: any
+
+  async componentDidMount () {
+    const { fioWallets } = this.props
+    if (fioWallets.length === 0) {
+      this.wallet = await this.createFioWallet()
+    } else {
+      this.wallet = fioWallets[0]
+    }
+    this.getRegexFromDomain(FIO_DOMAIN_DEFAULT)
+  }
+
+  createFioWallet = async () => {
```

Comment:
Add return type. Many other methods should also have return type added.

Reply:
ok